### PR TITLE
Take variant id of the network into account when searching equipments in ElasticSearch

### PR DESCRIPTION
**Pull request by niconoir**

### CODE REVIEW

File: src/test/java/org/gridsuite/study/server/StudyServiceTests.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Nicolas Noir <nicolas.noir at rte-france.com>
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE, properties = {"spring.data.elasticsearch.enabled=true"})
+public class StudyServiceTests {
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/study/server/StudyServiceTests.java:39]: Maybe rename into StudySearchServiceTests ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/study/server/StudyServiceTests.java (hunk 1 above)
Comment (niconoir) [reply src/test/java/org/gridsuite/study/server/StudyServiceTests.java:39]: StudyServiceSearchTests would be ok ? (searching using StudyService)

Comment (niconoir) [reply src/test/java/org/gridsuite/study/server/StudyServiceTests.java]: Done

**Review by FranckLecuyer — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Comment by SlimaneAmar** [comment]
Remova all StreamSupport.stream ?

File: src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java [hunk 2]
```diff
@@ -30,30 +31,45 @@
 
     private final EquipmentInfosRepository equipmentInfosRepository;
 
+    private final TombstonedEquipmentInfosRepository tombstonedEquipmentInfosRepository;
+
     private final ElasticsearchOperations elasticsearchOperations;
 
-    public EquipmentInfosServiceImpl(EquipmentInfosRepository equipmentInfosRepository, ElasticsearchOperations elasticsearchOperations) {
+    public EquipmentInfosServiceImpl(EquipmentInfosRepository equipmentInfosRepository, TombstonedEquipmentInfosRepository tombstonedEquipmentInfosRepository, ElasticsearchOperations elasticsearchOperations) {
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java:39]: Done

File: src/main/java/org/gridsuite/study/server/dto/TombstonedEquipmentInfos.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ */
+@SuperBuilder
+@NoArgsConstructor
+@Getter
+@ToString
+@EqualsAndHashCode
+@Document(indexName = "#{@environment.getProperty('index.prefix')}tombstoned-equipments")
+@Setting(settingPath = "elasticsearch_settings.json")
+@TypeAlias(value = "TombstonedEquipmentInfos")
+public class TombstonedEquipmentInfos {
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/dto/TombstonedEquipmentInfos.java]: Done

File: src/main/java/org/gridsuite/study/server/elasticsearch/TombstonedEquipmentInfosRepository.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.util.UUID;
+
+/**
+ * @author Nicolas Noir <nicolas.noir at rte-france.com>
+ */
+@ConditionalOnExpression("'${spring.data.elasticsearch.enabled:false}' == 'true'")
+@Lazy
+public interface TombstonedEquipmentInfosRepository extends ElasticsearchRepository<TombstonedEquipmentInfos, String> {
+    Iterable<TombstonedEquipmentInfos> findAllByNetworkUuid(@NonNull UUID networkUuid);
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/elasticsearch/TombstonedEquipmentInfosRepository.java:23]: It is used for unit tests

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**